eclipse/src/saros/editor/annotations/SarosAnnotation.java

```diff
@@ -73,31 +78,46 @@ public static Color getUserColor(int colorID) {
   }
 
   /**
-   * Returns the color that corresponds to a user.
+   * Returns the color for the given annotation type and user color id.
    *
    * <p><b>Important notice:</b> Every returned color instance allocates OS resources that need to
    * be disposed with {@link Color#dispose()}!
    *
    * @param type
-   * @param colorID
+   * @param colorId
    * @return the corresponding color or a default one if no color is stored
    */
-  protected static Color getColor(String type, int colorID) {
-
-    type = getNumberedType(type, colorID);
+  public static Color getColor(final String type, final int colorId) {
 
-    AnnotationPreferenceLookup lookup = EditorsUI.getAnnotationPreferenceLookup();
+    final String typeToLookup = getNumberedType(type, colorId);
 
-    AnnotationPreference ap = lookup.getAnnotationPreference(type);
+    final AnnotationPreference ap =
+        EditorsUI.getAnnotationPreferenceLookup().getAnnotationPreference(typeToLookup);
 
     if (ap == null)
-      throw new RuntimeException(
-          "could not read color value of annotation '" + type + "' because it does not exists");
+      throw new IllegalArgumentException(
+          "could not read color value of annotation '"
+              + typeToLookup
+              + "' because it does not exists");
+
+    RGB rgb =
+        PreferenceConverter.getColor(EditorsUI.getPreferenceStore(), ap.getColorPreferenceKey());
 
-    return new Color(Display.getDefault(), ap.getColorPreferenceValue());
+    if (rgb == PreferenceConverter.COLOR_DEFAULT_DEFAULT) { // NOPMD
+      rgb = ap.getColorPreferenceValue();
+    }
+
+    if (rgb == null)
+      throw new IllegalArgumentException(
+          "annotation '" + typeToLookup + "' does not have a default color");
+
+    return new Color(Display.getDefault(), rgb);
   }
 
-  /** Loads the colors from the plugin.xml and overwrites possible errors in the PreferenceStore */
+  /**
+   * Loads the colors from the plugin.xml and overwrites possible errors in the EditorsUI preference
```

Comment:
```suggestion
   * Loads the default Saros colors from the plugin.xml and overwrites possible errors in the EditorsUI preference
```

Reply:
Smurfing code with Saros is not always the best option. It should be clear that this is the plugin.xml of the Saros plugin. The class is already smurfed with Saros.